### PR TITLE
Allow run_id to be referenced from resources when invoking

### PR DESCRIPTION
**Pull request by dpeng817**
Relevant to https://github.com/dagster-io/dagster/issues/6091

### CODE REVIEW

File: python_modules/dagster/dagster/core/execution/context/invocation.py [hunk 1]
```diff
@@ -52,9 +52,11 @@ def __init__(
         resources_config: Dict[str, Any],
         instance: Optional[DagsterInstance],
         partition_key: Optional[str],
+        run_id: str,
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/core/execution/context/invocation.py:55]: if were going to add this, should we just take the run object instead here? [1]

**Comment by alangenfeld** [comment]
Should we be more thorough and have a sibling to `InitResourceContext` that is for direct invocation similar to the set up for the solid contexts? Im not sure how much that actually helps, but the `pipeline_run` comments on `build_resources` don't feel great.

The other thing we could do is just thread the `EPHEMERAL` through and not mess with the public API yet

**Review by alangenfeld — CHANGES_REQUESTED** [review]
to your queue for thoughts

**Comment by dpeng817** [comment]
Closing for now until I have time to revisit

**Pull request closed without merging**